compiler/codegen.js

```diff
@@ -3950,6 +3943,23 @@ const generateForOf = (scope, decl) => {
     ...number(0, Valtype.i32),
     [ Opcodes.local_set, counter ],
 
+    // check tmp is iterable
```

Comment:
Is this needed? Won't it will just fail the typeswitch and error?

Reply:
It's because of the length check beforehand that things can fail, and because most memory is zero in Porffor, that's where it fails. I'm honestly really surprised that it doesn't cause a runtime error more frequently